### DISCUSSION TITLE
Hardware error in booting Linux || Polarfire Discovery Kit

### DISCUSSION THREAD

**Discussion by the-systematic-chaos** [polarfire-soc/polarfire-soc-documentation#492] [Linux build systems for PolarFire SoC]
I was trying to boot linux on the polarfire discovery kit. I used the latest image from the github and the job file as well. However, it keeps running into a hardware frame error when it starts to boot up. The SD card is a SanDisk 256G SD Card. Are there any pointers or any steps that I am missing here? Any help would be appreciated. Thanks.

https://github.com/polarfire-soc/polarfire-soc-discovery-kit-reference-design/releases/tag/2024.04
![hw_error](https://github.com/user-attachments/assets/c76515e3-d63d-4a4a-8f9b-16e6ecbbe385)
![job_file](https://github.com/user-attachments/assets/f5fcb0b4-6e8a-4b6b-b874-65aa29887998)

**Comment by hughbreslin** [comment]
Hi @the-systematic-chaos, it looks like you have your baud rate set to 9600, can you retry with 115200?

**Reply by the-systematic-chaos** [reply]
Hi I have done what you suggested. It seems I am getting  few logs but the hardware error is still persistent.

![jobfile_serialcom](https://github.com/user-attachments/assets/06752254-a527-4840-ab35-9c5f0f5c3c47)
![jobfile_serialcom2](https://github.com/user-attachments/assets/daac83d8-9f32-4e67-ad10-47f519653a77)
![jobfile_serialcom3](https://github.com/user-attachments/assets/06476208-3f28-41a5-9b2a-ed1d40f7d06d)
[device logs.txt](https://github.com/user-attachments/files/16856643/device.logs.txt)
[device logs2.txt](https://github.com/user-attachments/files/16856644/device.logs2.txt)

These are the logs. Some of these aren't reproduceable, the serial ports sometimes shuts down after this error occurs and I can't seem to get a good screenshot in time.

**Reply by hughbreslin** [reply]
Hmm ok thats odd, what app are you using to capture the UART output? I normally use teraterm or screen on linux and I've never seen this behavior before, could you confirm your configuration matches the recommended settings? "Serial port settings: 115220 baud, no flow control, no parity" from the [user guide](https://github.com/polarfire-soc/polarfire-soc-documentation/blob/master/reference-designs-fpga-and-development-kits/mpfs-discovery-kit-user-guide.md#coming-out-of-the-box).

**Reply by Bossman5960** [reply]
I'm getting a similar error:
```
Starting kernel ...

[12.748165] u54_1: BEU event:              Load or store TileLink bus error
[12.757045] u54_1: BEU event:              Load or store TileLink bus error
```
That last line repeats until I close the connection. I've attached the complete serial output below.

[yocto error.txt](https://github.com/user-attachments/files/16949961/yocto.error.txt)

To reproduce the **TileLink Bus Error (BEU)** on the PolarFire SoC Discovery Kit:
1. Flash the board using **FlashPro Express** with the **MPFS_DISCOVERY.job** file extracted from **MPFS_DISCOVERY.zip** found at [this github](https://github.com/polarfire-soc/polarfire-soc-discovery-kit-reference-design/releases)
2. Flash a micro SD card with **core-image-minimal-dev-mpfs-video-kit-20240724114657.rootfs.wic** extracted from **core-image-minimal-dev-mpfs-video-kit-20240724114657.rootfs.wic.gz** found at [this github](https://github.com/polarfire-soc/meta-polarfire-soc-yocto-bsp/releases). 
3. After inserting the SD card and powering on the board, connect via UART, and observe the kernel boot process, where repeated **TileLink bus errors** occur, indicating a memory access violation during the transition from the bootloader to the kernel.

**Reply by hughbreslin** [reply]
Hey @Bossman596 I don't think these are connected, can you try using the disco kit yocto image instead of the video kit one? The image is available in the [2024.06 release](https://github.com/polarfire-soc/meta-polarfire-soc-yocto-bsp/releases/tag/v2024.06) - the video kit is a different kit with a different configuration so that would explain why its Linux image wouldn't work
